ckanext/geocat/metadata.py

```diff
@@ -408,6 +421,27 @@ def get_metadata(self, xml):
 
         return distributions
 
+    # Use the original dist as template to create a new dist.
+    # Also set the url to the access_url (and the download_url if it exists)
+    # and delete the url_list on the copy as we don't need it afterwards.
+    def _create_dist_copy(self, orig_dist, access_url, download_url=None):
+        # type: (object, StringValue, StringValue) -> object
+        dist = orig_dist.copy()
+        dist['url'] = access_url
+        if download_url:
```

Comment:
I would place the "download" special case in the `GeocatDcatDownloadDistributionMetdata` class